### PR TITLE
Add missing tests for scalar corner cases

### LINKED ISSUE #173: Add tests for scalar corner cases in @system
See https://github.com/JuliaReach/MathematicalSystems.jl/pull/168#issue-375862105.

Add tests after #168 is merged.

### PR DESCRIPTION
**Pull request by ueliwechsler**
closes #173 
and failing test work (it failed due to resolving #169)

### CODE REVIEW

File: test/@system.jl [hunk 1]
```diff
@@ -252,7 +259,7 @@ end
     @test @system(x' = -1.0x, x(0) ∈ Interval(-1, 1)) == s
 
     # similar for integers
-    s = IVP(LinearDiscreteSystem(I(-1, 1)), [1])
+    s = IVP(LinearDiscreteSystem(I(-1., 1)), [1])
     @test @system(x⁺ = -x, x(0) ∈ [1]) == s
```
Comment (schillic) [inline test/@system.jl]: I think the integer was on purpose (see the comment).

Comment (ueliwechsler) [reply test/@system.jl]: Can you point me to the comment you are referring to?

Comment (schillic) [reply test/@system.jl]: the first line in the code excerpt :wink:

Comment (ueliwechsler) [reply test/@system.jl]: 🙈

Comment (ueliwechsler) [reply test/@system.jl]: Actually, I don't think we should have this distinction in the test for `IVPs` because this has nothing to do with the creation of a `IVP` but with the design decision from #139

Comment (ueliwechsler) [reply test/@system.jl]: https://github.com/JuliaReach/MathematicalSystems.jl/blob/d9e9331fb6add240931f92cd88e82948750fc8a1/src/macros.jl#L722-L724

Comment (schillic) [reply test/@system.jl]: Good catch. Then we should remove the comment and check that the type is indeed `Float64` I think.

Comment (ueliwechsler) [reply test/@system.jl]: Like this?

Comment (schillic) [reply test/@system.jl]: Not quite because previously the test also worked, probably because `==` converts automatically. Maybe something like (not tested)
```julia
    s = @system(x⁺ = -x, x(0) ∈ [1])
    @test s == IVP(LinearDiscreteSystem(I(-1.0, 1)), [1]) && eltype(s.A) == Float64
```

Comment (schillic) [reply test/@system.jl]: Note that you need to swap the definition of `s` for this test to work.

Comment (ueliwechsler) [reply test/@system.jl]: Sure, I tried to do a hasty quick fix. I am on it

Comment (schillic) [reply test/@system.jl]: And also `s` does not have a field `A` (my bad).

Comment (schillic) [reply test/@system.jl]: > Note that you need to swap the definition of s for this test to work.

I fixed that.

**Comment by mforets** [comment]
Thanks!
